### ISSUE TITLE
Code Action Idea: Reorder If/Else Branches

### ISSUE THREAD

**Issue by hediet** [p42ai/refactor-vscode#28]
```ts
if (cond1) {
    body1
} el|se if (cond2) {
    body2
} else {
    body3
}
``` 
(| cursor)

-- Move Item Up ->
```ts
|if (cond2) {
    body2
} else if (cond1) {
    body1
} else {
    body3
}
```

Could use the same shortcut for reordering parameters/members etc.

**Comment by lgrammel** [comment]
Great idea! I'm thinking about introducing a "move" class of refactorings for reordering and moving parts of the code without changing it. This could fall into that category.

**Comment by lgrammel** [comment]
I've introduced a "swap if-else-if branches" refactoring in P42 `v1.131.0`:

![action-swap-if-else-if-branches](https://user-images.githubusercontent.com/205036/181273817-4bdca21c-4984-49a2-9061-5ca5f4cd4250.gif)

Thanks again for hte refactoring idea!

**Comment by hediet** [comment]
Nice work!

What does swap branches do exactly though?

I think a reactoring "move branch up" and "move branch down" would be more intuitive...

**Comment by hediet** [comment]
> in between elements: swap

This I don't understand. Why not move down/up here too?

**Comment by lgrammel** [comment]
My current interpretation is that when you are on the "else", you are not on an if element, so it is not clear what you would move "down". But this might be too technical of a notion (based on the AST). I'll think a bit more about this - I agree that having an intuitive and unified system is desirable.

**Comment by hediet** [comment]
> Here is my current thinking:

I wouldn't distinguish between `el|se if` and `else i|f`.

I wonder what `swap` would do on the last `else` in start3?
Personally, I find the description "Move branch 3" up easier than "Swap branch 3 (with branch 2)".

**Comment by lgrammel** [comment]
> I wonder what `swap` would do on the last `else` in start3? Personally, I find the description "Move branch 3" up easier than "Swap branch 3 (with branch 2 or branch 1?)".

This is the current implementation:

Start 4:
```ts
if (cond1) {
    body1
} else if (cond2) {
    body2
} el|se if (cond3) {
    body3
}
```

Start 4 --- Swap -->

```ts
if (cond1) {
    body1
} else if (cond3) {
    body3
} else if (cond2) {
    body2
}
```

One challenge is the action label - to distinguish the branches in the action label, I could include e.g. the condition, but this will lead to issues with long conditions.

**Comment by hediet** [comment]
I think swap is also a more mentally complex operation: "Swap" exchanges the upper element with the lower element - so mentally, you have to think about both elements.

However, "move up" focuses only on the current element.

Effectively, they would do the same though.

**Comment by lgrammel** [comment]
Yeah, I agree that's a good point. My main concern is the ambiguity of what up/down means for the space in between two elements. I'll think about it - for e.g. statements up/down is the best option for sure, and for if/else I could add it as well with different activation zones. I'll re-open this issue to keep track of it.

**Comment by hediet** [comment]
>  My main concern is the ambiguity of what up/down means for the space in between two elements. I'll think about it

Ask some users what they would expect to happen with move up/down when the cursor is on the else branch.

**Comment by lgrammel** [comment]
I changed the refactoring to move up / move down:

![action-move-if-else-if-branches](https://user-images.githubusercontent.com/205036/182602063-57aff039-a5f9-40e2-a115-9e95a4fc1d94.gif)

Available in `v1.132.0`
